### PR TITLE
Associations: changes

### PR DESCRIPTION
**Pull request by Paul-tes**
- Added a class Classroom with the following:
   - @label instance variable, that initialized in the constructor.
   - Setter and getter for @label.
- Added the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
   - Added  has-many side (a classroom has many students).
   - Added belongs-to side (a student belongs to a classroom).
   - Added **add_student** method when adding a student to a classroom it also sets the classroom for the student.
   - Added setter when setting the classroom for a student it also adds it to the classrooms' students.
- Added a class Book with the following:
   - @title and @author instance variables, should be initialized in the constructor.
   - Added Setters and getters for instance variables.
- Added a class Rental with the following:
   - @date instance variable, should be initialized in the constructor.
   - Added Setter and getter for @date (remember about attr_accessor).
- Added the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
   - Added the has-many side of Book and Rental (a book has many rentals).
   - Added the belongs-to side of Rental and Book (a rental belongs to a book).
   - Added the has-many side of Person and Rental (a person has many rentals).
   - Added the belongs-to side of Rental and Person (a rental belongs to a person).
   - Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (nahnahdev) [inline book.rb]: - [x] Great job so far @Paul-tes   You have done an impressive job so far. However, per the project requirements, you needed to:

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
> Create the has-many side of Book and Rental (a book has many rentals)

- Currently, this isn't being implemented properly in the `Book class`. I will suggest you ensure that the book itself is sent as a parameter to create a rental thereby achieving the `has-many` association. :wink: 
- Here is a [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship%20def%20add_rental(person,%20date)) to an example of this implementation in a previous lesson :+1: 
- Example implementation

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
   def initialize(age, parent_permission: true, name: 'Unknown')
     super()
     @id = Random.rand(1..10_000)
     @parent_permission = parent_permission
     @age = age
     @name = name
+    @rentals = []
   end
 
   def can_use_services?
```
Comment (nahnahdev) [inline person.rb]: - [x] Just as I mentioned in my previous comment, a similar implementation should be present in this `Person class` as well :+1: 
- [x]  Per the project requirements, you needed to:

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
> Create the has-many side of Person and Rental (a person has many rentals).

- Currently, this isn't being implemented in the `Person class`. I will suggest you include `a method for adding rentals `in the class. The person itself should be sent as a parameter to create the new rental thereby achieving the `has-many` association. :wink: 
- Here is a [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship%20def%20add_rental(person,%20date)) to an example of this implementation in a previous lesson :+1:

File: student.rb [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 
 class Student < Person
-  def initialize(age, classroom, parent_permission: true, name: 'Unknown')
+  attr_accessor :classroom
+
+  def initialize(age, parent_permission: true, name: 'Unknown')
     super(age, parent_permission: parent_permission, name: name)
-    @classroom = classroom
+    @classroom = nil
   end
```
Comment (nahnahdev) [inline student.rb]: - [x] Lastly, In the `student.rb` file, you need to ensure that there is a `belongs-to relationship` between `Classroom` and `Student` :+1:  You did an incredible job creating the `'has-many' side (a classroom has many students)` so kindly create a method that serves as the `'belongs-to' side (a student belongs to a classroom)` :wink: 
- [x] That way, when setting the classroom for a student it also adds it to the classrooms' students :+1: 

- Example implementation 
```ruby
class Student < Person
  attr_reader :classroom
  
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end
```

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @Paul-tes   , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Professional README :heavy_check_mark: 
- All required files present :heavy_check_mark: 
- Associations almost completed :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @Paul-tes , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**